workflows/sarek/main.nf

```diff
@@ -149,8 +156,59 @@ workflow SAREK {
 
         // Figure out if input is bam or fastq
         input_sample_type = input_sample.branch{
-            bam:   it[0].data_type == "bam"
-            fastq: it[0].data_type == "fastq"
+            bam:                 it[0].data_type == "bam"
+            fastq_gz:            it[0].data_type == "fastq_gz"
+            one_fastq_gz_spring: it[0].data_type == "one_fastq_gz_spring"
+            two_fastq_gz_spring: it[0].data_type == "two_fastq_gz_spring"
+        }
+
+        // Two fastq.gz-files
+        fastq_gz = input_sample_type.fastq_gz.map { meta, files ->
+            def CN = params.seq_center ? "CN:${params.seq_center}\\t" : ''
+
+            // Here we're assuming that fastq_1 and fastq_2 are from the same flowcell:
+            def flowcell = flowcellLaneFromFastq(files[0])
+            // TO-DO: Would it perhaps be better to also call flowcellLaneFromFastq(files[1]) and check that we get the same flowcell-id?
+
+            // Don't use a random element for ID, it breaks resuming
+            def read_group = "\"@RG\\tID:${flowcell}.${meta.sample}.${meta.lane}\\t${CN}PU:${meta.lane}\\tSM:${meta.patient}_${meta.sample}\\tLB:${meta.sample}\\tDS:${params.fasta}\\tPL:${params.seq_platform}\""
+            meta  = meta - meta.subMap('lane') + [read_group: read_group.toString()]
+            return [ meta, files ]
+        }
+
+        // Just one fastq.gz.spring-file with both R1 and R2
+        fastq_gz_pair_from_spring = SPRING_DECOMPRESS_TO_FQ_PAIR(input_sample_type.one_fastq_gz_spring)
+
+        one_fastq_gz_from_spring = fastq_gz_pair_from_spring.fastq.map { meta, files ->
+            def CN = params.seq_center ? "CN:${params.seq_center}\\t" : ''
+
+            // Here we're assuming that fastq_1 and fastq_2 are from the same flowcell:
+            def flowcell = flowcellLaneFromFastq(files[0])
+            // TO-DO: Would it perhaps be better to also call flowcellLaneFromFastq(files[1]) and check that we get the same flowcell-id?
+
+            // Don't use a random element for ID, it breaks resuming
+            def read_group = "\"@RG\\tID:${flowcell}.${meta.sample}.${meta.lane}\\t${CN}PU:${meta.lane}\\tSM:${meta.patient}_${meta.sample}\\tLB:${meta.sample}\\tDS:${params.fasta}\\tPL:${params.seq_platform}\""
+            meta  = meta - meta.subMap('lane') + [read_group: read_group.toString()]
+            return [ meta, files ]
+        }
+
+        // Two fastq.gz.spring-files - one for R1 and one for R2
+        R1_fastq_gz_from_spring = SPRING_DECOMPRESS_TO_R1_FQ_PAIR(input_sample_type.two_fastq_gz_spring.map{ meta, files -> [meta, files[0] ]})
```

Comment:
is it right that this called for two_fastq_gz?

Reply:
I'm not too happy with the names of the "module instances" and channels, so fill free to improve on that. However, I think that the functionality here is as desired. The `.map{ meta, files -> [meta, files[0] ] }` should ensure that `SPRING_DECOMPRESS_TO_R1_FQ_PAIR` is called on the R1-fastq-gz-spring file.

Reply:
can the R1 / R2 channels be all lowercased?

Reply:
> can the R1 / R2 channels be all lowercased?

You mean R1/R2 should be lowercase in the variabel names like `R1_fastq_gz_from_spring`, right? That is fine with me, but we wanna keep R1/R2 uppercase in "module instances" like `SPRING_DECOMPRESS_TO_R1_FQ_PAIR`, right?